lib/sharepoint/client.rb

```diff
@@ -11,8 +11,80 @@ module Sharepoint
   class Client
     FILENAME_INVALID_CHARS = '~"#%&*:<>?/\{|}'
 
+    attr_accessor :token
+
+    class InvalidTokenError < StandardError
+    end
+
+    class Token
+      attr_accessor :expires_in
+      attr_accessor :access_token
+      attr_accessor :fetched_at
+      attr_reader :config
+
+      def initialize(config)
+        @config = config
+      end
+
+      def ensure
```

Comment:
ensure is a keyword, any chance to use a different name?

Reply:
Good point, I think we can easily use any other name

Reply:
right!